core/src/main/java/io/grpc/transport/AbstractServerStream.java

```diff
@@ -141,9 +141,9 @@ protected final void deframeFailed(Throwable cause) {
   }
 
   @Override
-  protected final void internalSendFrame(WritableBuffer frame, boolean endOfStream) {
+  protected final void internalSendFrame(WritableBuffer frame, boolean endOfStream, boolean flush) {
     if (frame.readableBytes() > 0) {
-      sendFrame(frame, false);
+      sendFrame(frame, false, endOfStream ? false : flush);
```

Comment:
In the case of bidi streaming, couldn't it be possible that server may not write again until it receives the next request from the client? In that case we would just want to pass `flush` directly here and not do anything special with `endOfStream`.